src/Tests/VCEL.Test/ComparisonOpTests.cs

```diff
@@ -302,6 +297,21 @@ public void CompareDecimalLeft(double a, object b, bool expected)
         public void CompareDecimalRight(double a, object b, bool expected)
             => Compare("B < A", expected, new { A = (decimal)a, B = b });
 
+        [TestCase("'ABC' ~ null")]
+        [TestCase("'ABC' matches null")]
+        [TestCase("'ABC' matches A + 'A'")]
+        [TestCase("'ABC' matches A + B")]
+        [TestCase("(null + '') ~ ('Y' + null)")]
+        [TestCase("'ABC' matches null")]
+        [TestCase("'ABC' ~ null")]
+        [TestCase("(null + '') ~ ('Y' + null)")]
+        [TestCase("'A' + 'BC' matches 'A' + '.*'")]
+        public void ParseShouldFail(string exprString)
```

Comment:
We are now limit the regex pattern to be static string.
These expr should be fail